doc/cardano-testnet-testing.md

```diff
@@ -0,0 +1,432 @@
+# CTL integration with Cardano Testnet
+
+**Table of Contents**
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Architecture](#architecture)
+- [Testing contracts](#testing-contracts)
+  - [Testing with Mote](#testing-with-mote)
+    - [Overview](#overview)
+    - [Using Mote testing interface](#using-mote-testing-interface)
+    - [Internal implementation overview](#internal-implementation-overview)
+  - [Testing in Aff context](#testing-in-aff-context)
+  - [Sharing wallet state between tests](#sharing-wallet-state-between-tests)
+  - [Writing checks in tests](#writing-checks-in-tests)
+  - [Note on SIGINT](#note-on-sigint)
+  - [Testing with Nix](#testing-with-nix)
+- [Cluster configuration options](#cluster-configuration-options)
+  - [Current limitations](#current-limitations)
+- [Using addresses with staking key components](#using-addresses-with-staking-key-components)
+- [Limitations](#limitations)
+- [See also](#see-also)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+## Architecture
+
+CTL depends on a number of binaries in the `$PATH` to execute tests on the
+Cardano Testnet:
+
+- [`cardano-node`](https://github.com/IntersectMBO/cardano-node) to connect to the Cardano Testnet
+- [`ogmios`](https://ogmios.dev/)
+- [`kupo`](https://cardanosolutions.github.io/kupo/)
+
+All of these are provided by CTL's `overlays.runtime` (and are provided in
+CTL's own `devShell`). You **must** use the `runtime` overlay or otherwise make
+the services available in your package set (e.g. by defining them within your
+own `overlays` when instantiating `nixpkgs`) as `purescriptProject.runTest`
+expects all of them; an example of using CTL's overlays is in the
+[`ctl-scaffold` template](../templates/ctl-scaffold/flake.nix#L35).
+
+The services are NOT run by `docker-compose` (via `arion`) as is the case with
+`launchCtlRuntime`: instead, they are started and stopped on each CTL
+`ContractTest` execution by CTL itself.
+
+If you have based your project on the [`ctl-scaffold`
+template](../templates/ctl-scaffold) then you have two options to run tests on
+the Cardano Testnet:
+1. `nix develop` followed by `npm run test` (recommended for development)
+2. `nix run .#checks.x86_64-linux.ctl-scaffold-local-testnet-test`
+   * where you'd usually replace `x86_64-linux` with the system you run tests on
+   * and `ctl-scaffold-local-testnet-test` with the name of the test derivation for your project;
+   * note that building of your project via Nix will fail in case there are any PureScript compile-time warnings.
```

Comment:
```suggestion
```

It will be obvious anyway if there are errors. No need to overload users with info before it's needed